packages/fluentui/react-northstar/src/components/Form/FormField.tsx

```diff
@@ -51,74 +57,121 @@ export interface FormFieldProps extends UIComponentProps, ChildrenComponentProps
+  setStart();
+
+  const {
+    children,
+    control,
+    id,
+    label,
+    message,
+    name,
+    required,
+    type,
+    className,
+    design,
+    styles,
+    variables,
+    inline,
+  } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(FormField.handledProps, props);
+
+  const getA11yProps = useAccessibility<never>(props.accessibility, {
+    debugName: FormField.displayName,
+    rtl: context.rtl,
+  });
+
+  const { classes, styles: resolvedStyles } = useStyles<FormFieldStylesProps>(FormField.displayName, {
+    className: formFieldClassName,
+    mapPropsToStyles: () => ({
+      type,
+      inline,
+      required,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const labelElement = Text.create(label, {
+    defaultProps: () => ({
+      as: 'label',
+      htmlFor: id,
+      styles: resolvedStyles.label,
+    }),
+  });
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false,
+  const messageElement = Text.create(message, {
+    defaultProps: () => ({
+      styles: resolvedStyles.message,
     }),
-    control: customPropTypes.itemShorthand,
-    id: PropTypes.string,
-    inline: PropTypes.bool,
-    label: customPropTypes.itemShorthand,
-    message: customPropTypes.itemShorthand,
-    name: PropTypes.string,
-    required: PropTypes.bool,
-    type: PropTypes.string,
-  };
+  });
 
-  static defaultProps = {
-    as: 'div',
-    control: { as: Input },
-  };
+  const controlElement = Box.create(control || {}, {
+    defaultProps: () => ({ required, id, name, type, styles: resolvedStyles.control }),
+  });
 
-  renderComponent({ ElementType, classes, accessibility, styles, unhandledProps }): React.ReactNode {
-    const { children, control, id, label, message, name, required, type } = this.props;
-
-    const labelElement = Text.create(label, {
-      defaultProps: () => ({
-        as: 'label',
-        htmlFor: id,
-        styles: styles.label,
-      }),
-    });
-
-    const messageElement = Text.create(message, {
-      defaultProps: () => ({
-        styles: styles.message,
-      }),
-    });
-
-    const controlElement = Box.create(control || {}, {
-      defaultProps: () => ({ required, id, name, type, styles: styles.control }),
-    });
-
-    const content = (
-      <>
-        {this.shouldControlAppearFirst() && controlElement}
-        {labelElement}
-        {!this.shouldControlAppearFirst() && controlElement}
-        {messageElement}
-      </>
-    );
-
-    return (
-      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
-        {childrenExist(children) ? children : content}
-      </ElementType>
-    );
-  }
-
-  shouldControlAppearFirst = () => {
-    const { type } = this.props;
+  const shouldControlAppearFirst = () => {
+    const { type } = props;
```

Comment:
```suggestion
```
You have `type` destructured on line 75, shouldn't it be enough?

Reply:
Yep, thanks :)